lib/prm/employees/employees.ex

```diff
@@ -44,6 +44,36 @@ defmodule PRM.Employees do
     |> preload_relations(params)
   end
 
+  def get_search_query(Employee = entity, changes) do
+    params =
+      changes
+      |> Map.delete(:tax_id)
+      |> Map.delete(:edrpou)
+      |> Map.to_list()
+
+    tax_id = Map.get(changes, :tax_id)
+    edrpou = Map.get(changes, :edrpou)
+
+    query = from e in entity
+    query = if tax_id do
+      query
+      |> join(:left, [e], p in assoc(e, :party))
+      |> where([e, p], p.tax_id == ^tax_id)
+    else
+      query
+    end
+
+    query = if edrpou do
+      query
+      |> join(:left, [e], le in assoc(e, :legal_entity))
+      |> where([e, le], le.edrpou == ^edrpou)
+    else
+      query
+    end
+
+    query |> where(^params)
+  end
```

Comment:
use pattern matching
`def query_tax_id(query, nil), do query`
`def query_tax_id(query, tax_id) do`
` . query`
` . |> join(:left, [e], p in assoc(e, :party))`
` . |> where([e, p], p.tax_id == ^tax_id)`
`end`
and in code:
`from e in entity`
`|> query_tax_id(tax_id)`
`|> query_edrpou(Map.get(changes, :edrpou))`